doc/api/test.md

```diff
@@ -603,6 +603,220 @@ test('mocks setTimeout to be executed synchronously without having to actually w
 });
 ```
 
+### Dates
+
+The mock timers API also allows the mocking of the `Date` object. This is a
+useful feature for testing time-dependent functionality, or to simulate
+internal calendar functions such as `Date.now()`.
+
+The dates implementation is also part of the [`MockTimers`][] class. Refer to it
+for a full list of methods and features.
+
+**Note:** Dates and timers are dependent when mocked together. This means that
+if you have both the `Date` and `setTimeout` mocked, advancing the time will
+also advance the mocked date as they simulate a single internal clock.
+
+The example below show how to mock the `Date` object and obtain the current
+`Date.now()` value.
+
+```mjs
+import assert from 'node:assert';
+import { test } from 'node:test';
+
+test('mocks the Date object', (context) => {
+  // Optionally choose what to mock
+  context.mock.timers.enable({ apis: ['Date'] });
+  // If not specified, the initial date will be based on 0 in the UNIX epoch
+  assert.strictEqual(Date.now(), 0);
+
+  // Advance in time will also advance the date
+  context.mock.timers.tick(9999);
+  assert.strictEqual(Date.now(), 9999);
+});
+```
+
+```js
```

Comment:
I feel like I already did those suggestions, any reason it was reverted / not applied?
````suggestion
```cjs
````